tensor_comprehensions/pybinds/tclib.cc

```diff
@@ -273,6 +273,23 @@ struct TcExecutor {
       return tupleOrTensor(convertToPyObjects(atOutputs));
     }
   }
+
+  size_t profile(const py::tuple& inputs, const py::tuple& outputs) {
+    if (outputs.size() > 0) {
+      auto atOutputs = getATenTensors(outputs);
+      auto atInputs = getATenTensors(inputs);
+      tc::ProfilingInfo profinfo =
+          tc::aten::profile(*executor, atInputs, atOutputs);
+      return profinfo.kernelRuntime.toMicroSeconds();
+    } else {
+      auto atInputs = getATenTensors(inputs);
```

Comment:
Most of these lines are duplicates of the "if" block and can be hoisted.  You should be able to just use a ternary operator when assigning `atOutputs` and remove the branch completely.